diff --git a/src/AsciiRenderer.java b/src/AsciiRenderer.java
new file mode 100644
index 0000000..ea17f09
--- /dev/null
+++ b/src/AsciiRenderer.java
@@ -0,0 +1,12 @@
+public class AsciiRenderer implements Renderer {
+
+    @Override
+    public String renderCircle(int radius) {
+        return "ASCII circle radius=" + radius;
+    }
+
+    @Override
+    public String renderSquare(int side) {
+        return "ASCII square side=" + side;
+    }
+}
\ No newline at end of file
diff --git a/src/Main.java b/src/Main.java
index 22ae859..fee170e 100644
--- a/src/Main.java
+++ b/src/Main.java
@@ -53,7 +53,27 @@ public class Main {
 
         passed += checkRuntimeSwitch();
 
-        System.out.println("SUMMARY: " + passed + "/5 PASS");
+        Circle circleAscii =
+                new Circle("C3", 2, new AsciiRenderer());
+
+        passed += check(
+                "T6",
+                "Circle + AsciiRenderer",
+                circleAscii.execute(),
+                "ASCII circle radius=2"
+        );
+
+        Square squareAscii =
+                new Square("S3", 3, new AsciiRenderer());
+
+        passed += check(
+                "T7",
+                "Square + AsciiRenderer",
+                squareAscii.execute(),
+                "ASCII square side=3"
+        );
+
+        System.out.println("SUMMARY: " + passed + "/7 PASS");
     }
 
     private static int checkRuntimeSwitch() {
